### ISSUE TITLE
Option to allow `fixup!` and `squash!` in commit messages

### ISSUE THREAD

**Issue by lobotmcj** [commitizen-tools/commitizen#414]
## Description
It would be nice if `cz check` didn't fail for fixup and squash commits, e.g., `git commit --fixup <rev>` or `git commit --squash <rev>`

## Possible Solution
Add an option for this and, if true, prepend the commit message check with something like `r"((fixup|squash)\!)?\s"`

**Comment by Lee-W** [comment]
I'm good with this feature @Woile What do you think?

**Comment by Yajo** [comment]
I don't think this is related to git itself. It's just a matter of allowing fixup commits in the regexp.

For example, having the commit-msg pre-commit hook installed, there should be a way to let this *not* fail:

```sh
> git commit -a --fixup 57a2044f1290cba0ca6530e6e77cd5da243e74b1
commitizen check.........................................................Failed
- hook id: commitizen
- exit code: 14

commit validation: failed!
please enter a commit message in the commitizen format.
commit "": "fixup! feat: allow easy working"
pattern: (build|ci|docs|feat|fix|perf|refactor|style|test|chore|revert|bump)(\(\S+\))?!?:(\s.*)
```

**Comment by Lee-W** [comment]
if that's the case, I'm ok with it. @Woile what do you think?

**Comment by Woile** [comment]
fine by me as well :+1:

**Comment by Lee-W** [comment]
Thanks @wombelix I think it's a great workaround

**Comment by Lee-W** [comment]
I think this is supported by default in the latest version. Let's close it. Thanks @bearomorphism  for reminding me
